ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/BlockManager.java

```diff
@@ -186,17 +223,18 @@ public void onBlockValidated(SignedBeaconBlock block) {
   }
 
   private void importBlockIgnoringResult(final SignedBeaconBlock block) {
-    doImportBlock(block, Optional.empty()).ifExceptionGetsHereRaiseABug();
+    doImportBlock(block, Optional.empty(), Optional.empty()).ifExceptionGetsHereRaiseABug();
   }
 
   private SafeFuture<BlockImportResult> doImportBlock(
       final SignedBeaconBlock block,
-      final Optional<BlockImportPerformance> blockImportPerformance) {
+      final Optional<BlockImportPerformance> blockImportPerformance,
+      final Optional<SafeFuture<BlockImportResult>> consensusValidation) {
```

Comment:
We need a better name for this variable to clarify that it acts as a "probe" or a callback of consensus validation rules.

Maybe something like `consensusValidationListener` or `onConsensusValidationCompleted`.